XAlloc.c

```diff
@@ -39,6 +41,32 @@ void* xRealloc(void* ptr, size_t size) {
    return data;
 }
 
+int xAsprintf(char** strp, const char* fmt, ...) {
+   va_list vl;
+   va_start(vl, fmt);
+   int _r = vasprintf(strp, fmt, vl);
+   va_end(vl);
+
+   if (_r < 0) {
+      fail();
```

Comment:
I do not like these `fail()` calls:
In case of an allocation error (or in the` xSnprintf()` case a too small buffer) we just exit cleanly, with no message or hint for the user **why** and no information for us developers **where**.
See also #115

Reply:
Replacing this call by some error handler shouldn't be an issue.

With the current base for this PR there's only `fail()` available as a sensible place to call, but replacing it with the functions introduced in #115 should be fine.

I'd prefer not touching this PR right now, as bugfixes remarked on in #115 are already addressed in #155 which this PR is based upon.